components/deleteResourceLink/index.jsx

```diff
@@ -39,18 +44,33 @@ export default function DeleteResourceLink({
   onDelete,
   ...linkProps
 }) {
-  const { data: resourceInfo, error: resourceInfoError } = useResourceInfo(
-    resourceIri
-  );
-  const { accessControl, error: accessControlError } = useAccessControl(
-    resourceInfo
-  );
+  const { fetch } = useSession();
 
-  const handleDelete = createDeleteHandler(accessControl, onDelete);
+  const { policiesContainer } = usePoliciesContainer();
+  const [policyUrl, setPolicyUrl] = useState(null);
 
-  if (resourceInfoError || accessControlError) {
-    return null;
-  }
+  useEffect(() => {
+    if (!policiesContainer || !resourceIri) return;
```

Comment:
You only use policyUrl for the createDeleteHandler, so I think we can defer some of the logic in the `useEffect` here by moving the getPolicyUrl to `deleteResource` in resource.js.